area/guadosalam.py

```diff
@@ -15,113 +16,74 @@
+                # Back out the door
+                FFXC.set_movement(0,-1)
+                memory.main.await_event()
+                checkpoint += 1
+            elif checkpoint == 8:
+                # Into the dining hall
+                FFXC.set_movement(0,1)
+                memory.main.await_event()
+                checkpoint += 1
+            elif pathing.set_movement(GuadoStart.execute(checkpoint)):
+                checkpoint += 1
+                logger.debug(f"Checkpoint {checkpoint}")
+        else:
+            FFXC.set_neutral()
             xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-    # Line up for Rikku/Yuna
-    while not pathing.set_movement([15, -52]):
-        pass
-
-    while not pathing.set_movement([22, -25]):
-        pass
-    logger.debug("Rikku conversation")
-    while memory.main.user_control():  # Start conversation with Rikku
-        pathing.set_movement([8, -26])
-        xbox.tap_b()
-    FFXC.set_neutral()
-    memory.main.click_to_control_3()
-
-    while not pathing.set_movement([27, -37]):
-        pass
-    logger.debug("Yuna conversation")
-    while memory.main.user_control():  # Yunas turn
-        pathing.set_movement([39, -33])
-        xbox.tap_b()
-    FFXC.set_neutral()
-    memory.main.click_to_control_3()
+    
+    # Checkpoint carries over.
+    # We skip the first movement checkpoint, replaced by the approach feature.
+    checkpoint += 1
+    logger.info("Now in the room with all the food")
+    while not memory.main.get_map() == 163:
+        if memory.main.user_control():
+            if checkpoint == 10:
+                if game_vars.csr():
+                    pathing.approach_party_member(Wakka)
+                else:
+                    pathing.approach_party_member(Auron)
+                checkpoint += 1
+            elif checkpoint == 11:
+                if game_vars.csr():
+                    pathing.approach_party_member(Lulu)
+                else:
+                    pathing.approach_party_member(Wakka)
+                checkpoint += 1
+            elif checkpoint == 12:
+                if game_vars.csr():
+                    pathing.approach_party_member(Auron)
+                    checkpoint = 15
+                else:
+                    pathing.approach_party_member(Lulu)
+                    checkpoint += 1
+            elif checkpoint == 18:
+                pathing.approach_party_member(Rikku)
+                checkpoint += 1
+            elif checkpoint == 20:
+                pathing.approach_party_member(Yuna)
+                checkpoint += 1
+            elif pathing.set_movement(GuadoStart.execute(checkpoint)):
+                checkpoint += 1
+                logger.debug(f"Checkpoint {checkpoint}")
+        else:
+            FFXC.set_neutral()
+            if memory.main.diag_skip_possible():
+                xbox.tap_b()
+            #elif memory.main.cutscene_skip_possible():
```

Comment:
That scene should be skippable, I don't know why it's not working. I need to come back later and figure out why, and then implement this again. Otherwise we lose a full minute during the run which is significant.

Reply:
oh I know what it is.

Reply:
And it won't work when loading into the game after Battle Site, it's directly connected to one of those scenes.